### PR TITLE
Issue #5124: remove usages of branchContains for DesignForExtensionCheck

### PR DESCRIPTION
**Pull request by rnveach**
Issue #5124

This is just a simple rewrite. Fixing code to use a valid Javadoc will be handled at https://github.com/checkstyle/checkstyle/issues/5450 .
Regression to come.

### CODE REVIEW

**Comment by rnveach** [comment]
Regression: http://rveach.no-ip.org/checkstyle/regression/reports/243/

No differences.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java [hunk 1]
```diff
@@ -169,8 +169,29 @@ private static boolean hasJavadocComment(DetailAST methodDef) {
      * @return true if a token has a javadoc comment.
      */
     private static boolean hasJavadocCommentOnToken(DetailAST methodDef, int tokenType) {
+        boolean result = false;
+
         final DetailAST token = methodDef.findFirstToken(tokenType);
-        return token.branchContains(TokenTypes.BLOCK_COMMENT_BEGIN);
+        DetailAST curNode = token;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java]: please move it to static function.

Should it be util method ? sounds line very likely to be reused.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java]: @rnveach , ping.
Lets finish this PR :) .

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java]: > please move it to static function

It already is in a static method.

> Should it be util method ?

How do you want to make this a util method? It has a result to assign, a condition to meet to assign the value, and an early end condition.

Once it is merged, `if (curNode.getType() == TokenTypes.BLOCK_COMMENT_BEGIN) {` has to be changed to fix https://github.com/checkstyle/checkstyle/issues/5450 .

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java]: hasJavadocCommentOnToken is specific to this Check method.
you can make `SomeUtil.branchContains(token, TokenTypes.BLOCK_COMMENT_BEGIN);`

ok ... if you plan to refactor it further in knonw place, ok to merge as is.

**Comment by romani** [comment]
ok to merge.

**Comment by romani** [comment]
@rnveach , is there any concerns ?

**Comment by romani** [comment]
@rnveach , let's merge this.
Undone item of this PR could be moved to separate issue.

I want to clean up list of open PRs, it is better to close stale, no interest PRs. Ones interest come back we can reopen PR or create new.

**Comment by rnveach** [comment]
I cannot approve my own PR, but I see no issues.

**Pull request merged**